metals/src/main/scala/scala/meta/internal/debug/DebugSession.scala

```diff
@@ -0,0 +1,63 @@
+package scala.meta.internal.debug
+
+import java.io.File
+import java.net.URI
+import java.nio.file.Files
+import java.nio.file.Paths
+import org.eclipse.lsp4j.debug.services.IDebugProtocolClient
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.internal.debug.protocol.LaunchParameters
+import scala.meta.internal.debug.protocol.OutputEventArguments._
+import scala.sys.process.Process
+import scala.sys.process.ProcessLogger
+
+final class DebugSession(
+    process: Process,
+    val exitCode: Future[Int]
+) {
+  def whenTerminated[A](f: Int => A)(implicit ec: ExecutionContext): Future[A] =
+    exitCode.map(f)
+}
+
+final class DebugSessionFactory(client: IDebugProtocolClient) {
+  def createCommand(params: LaunchParameters): Array[String] = {
```

Comment:
Does this need to be public? Maybe let's do it consistent like `createWorkingDir`